src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -158,6 +158,12 @@ var DOMPropertyOperations = {
         } else if (propertyInfo.hasBooleanValue ||
                    (propertyInfo.hasOverloadedBooleanValue && value === true)) {
           node.setAttribute(attributeName, '');
+        } else if (attributeName === 'value' && node.hasAttribute('value')) {
+          // Use loose coercion to prevent replacement on comparisons like
+          // '3e1' == 30 in Chrome (~52).
+          if (node.value != value) { // eslint-disable-line
+            node.setAttribute(attributeName, '' + value);
```

Comment:
So, there's a difference between setting `node.value` and calling `node.setAttribute('value')`.  `node.value` tracks the value of the node in JS, and `setAttribute` will actually set the attribute on the element in the DOM.

This change causes an issue where the DOM attribute is never updated, because `node.value` is already updated prior to this check, so it never actually sets the attribute.